packages/grid/src/vaadin-grid-selection-column-mixin.js

```diff
@@ -178,18 +189,45 @@ export const GridSelectionColumnMixin = (superClass) =>
     }
 
     /**
-     * Assuming the grid uses an items array data provider, fetches all the filtered items
-     * from the data provider and invokes the callback with the resulting array.
+     * Returns all items that the user may change the selection state of, based
+     * on the filter.
      *
      * @private
      */
-    __withFilteredItemsArray(callback) {
+    __getSelectableItems() {
+      // Assuming the grid uses an items array data provider, fetches all the
+      // filtered items from the data provider
+      let filteredItems = [];
       const params = {
         page: 0,
         pageSize: Infinity,
         sortOrders: [],
         filters: this._grid._mapFilters(),
       };
-      this._grid.dataProvider(params, (items) => callback(items));
+      this._grid.dataProvider(params, (items) => {
+        filteredItems = items;
+      });
```

Comment:
This assumes that the array data provider is always synchronous. Seems to be fine, I couldn't see a reason to make this work with async. data providers as well.

Reply:
This is obsolete now.